perfkitbenchmarker/windows_virtual_machine.py

```diff
@@ -0,0 +1,306 @@
+    self.temp_dir = None
+
+  def RemoteCommand(self, command, should_log=False, ignore_failure=False,
+                    suppress_warning=False):
+    """Runs a command on the VM.
+
+    Args:
+      command: A valid bash command.
+      should_log: A boolean indicating whether the command result should be
+          logged at the info level. Even if it is false, the results will
+          still be logged at the debug level.
+      ignore_failure: Ignore any failure if set to true.
+      suppress_warning: Suppress the result logging from IssueCommand when the
+          return code is non-zero.
+
+    Returns:
+      A tuple of stdout and stderr from running the command.
+
+    Raises:
+      RemoteCommandError: If there was a problem issuing the command.
+    """
+    set_error_pref = '$ErrorActionPreference="Stop"'
+
+    password = self.password.replace("'", "''")
+    create_cred = (
+        '$pw = convertto-securestring -AsPlainText -Force \'%s\';'
+        '$cred = new-object -typename System.Management.Automation'
+        '.PSCredential -argumentlist %s,$pw' % (password, self.user_name))
+
+    create_session = (
+        '$session = New-PSSession -Credential $cred -Port %s -ComputerName %s' %
+        (self.winrm_port, self.ip_address))
+
+    invoke_command = (
+        'Invoke-Command -Session $session -ScriptBlock { %s };'
+        'exit Invoke-Command -Session $session -ScriptBlock '
+        '{ $LastExitCode }' % command)
+
+    cmd = ';'.join([set_error_pref, create_cred,
+                    create_session, invoke_command])
+
+    stdout, stderr, retcode = vm_util.IssueCommand(
+        ['powershell', '-Command', cmd], timeout=None,
+        suppress_warning=suppress_warning, force_info_log=should_log)
+
+    if retcode and not ignore_failure:
+      error_text = ('Got non-zero return code (%s) executing %s\n'
+                    'Full command: %s\nSTDOUT: %sSTDERR: %s' %
+                    (retcode, command, cmd, stdout, stderr))
+      raise errors.VirtualMachine.RemoteCommandError(error_text)
+
+    return stdout, stderr
+
+  def RemoteCopy(self, local_path, remote_path='', copy_to=True):
+    """Copies a file to or from the VM.
+
+    Args:
+      local_path: Local path to file.
+      remote_path: Optional path of where to copy file on remote host.
+      copy_to: True to copy to vm, False to copy from vm.
+
+    Raises:
+      RemoteCommandError: If there was a problem copying the file.
+    """
+    if ':' in remote_path:
+      remote_path = remote_path.split(':', 1)[1]
+
+    set_error_pref = '$ErrorActionPreference="Stop"'
+
+    password = self.password.replace("'", "''")
+    create_cred = (
+        '$pw = convertto-securestring -AsPlainText -Force \'%s\';'
+        '$cred = new-object -typename System.Management.Automation'
+        '.PSCredential -argumentlist %s,$pw' % (password, self.user_name))
+
+    psdrive_name = self.name
+    root = '\\\\%s\\c$' % self.ip_address
```

Comment:
Done - now we default to the drive that was part of the path passed to copy file - if that didn't have a drive, then we use the system drive environment variable that we got during startup.